x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview_header/engine_overview_header.tsx

```diff
@@ -24,7 +24,7 @@ export const EngineOverviewHeader: React.FC<IEngineOverviewHeaderProps> = ({
     fill: true,
     iconType: 'popout',
     'data-test-subj': 'launchButton',
-  };
+  } as any;
```

Comment:
In Enterprise Search we use the shared [interface](https://github.com/elastic/ent-search/blob/master/app/javascript/shared/types.ts#L34) `IObject` for these generic objects. I wonder if something like that would male the linter happy?

Reply:
TBH I should probably be using `as EuiButtonProps` here, but when I do it I get this arcane semi-gibberish error that I can't figure out :dead_inside:

This is the only instance of me using `as any` in the plugin source code however so it's probably worth fixing or pairing with an EUI dev on 🤔

Reply:
Here's the error(s) when I use `as EuiButtonProps` for what it's worth:

```
Property 'href' does not exist on type 'EuiButtonProps'.
Property 'target' does not exist on type 'EuiButtonProps'.
Property 'onClick' does not exist on type 'EuiButtonProps'.
```

Which is weird because `href` and `onClick` definitely exist in the props tabs of https://elastic.github.io/eui/#/navigation/button 🤷

Reply:
Hmmm, maybe I can define a custom interface instead?

```
interface IEuiButtonProps extends EuiButtonProps {
  href: string;
  target: string;
  onClick(): void;
}
```

I think that would work actually! 🤔